### PR TITLE
Uast: proper resolve for local variables calls as functions (KT-25524)

### PR DESCRIPTION
**Pull request by xiexed**

### CODE REVIEW

File: plugins/uast-kotlin/src/org/jetbrains/uast/kotlin/expressions/KotlinUFunctionCallExpression.kt [hunk 1]
```diff
@@ -101,7 +106,29 @@ class KotlinUFunctionCallExpression(
     }
 
     override val receiver: UExpression?
-        get() = (uastParent as? UQualifiedReferenceExpression)?.takeIf { it.selector == this }?.receiver
+        get() {
+            (uastParent as? UQualifiedReferenceExpression)?.let { return it.takeIf { it.selector == this }?.receiver }
```
[the remark this answers is not in the document]
Comment (xiexed) [reply plugins/uast-kotlin/src/org/jetbrains/uast/kotlin/expressions/KotlinUFunctionCallExpression.kt]: it is already `let` with `if`, isnt?)

[the remark this answers is not in the document]
Comment (xiexed) [reply plugins/uast-kotlin/src/org/jetbrains/uast/kotlin/expressions/KotlinUFunctionCallExpression.kt]: you mean?
```
(uastParent as? UQualifiedReferenceExpression)?.let { if (it.selector == this) return it.receiver }
```

[the remark this answers is not in the document]
Comment (yanex) [reply plugins/uast-kotlin/src/org/jetbrains/uast/kotlin/expressions/KotlinUFunctionCallExpression.kt]: Exactly.

[the remark this answers is not in the document]
Comment (xiexed) [reply plugins/uast-kotlin/src/org/jetbrains/uast/kotlin/expressions/KotlinUFunctionCallExpression.kt]: Updated. actually it has changed behaviour a bit... but probably it is ok

**Comment by xiexed** [comment]
merged manually

**Pull request closed without merging**